### PR TITLE
Optimize some Mutect-related tools

### PR DESCRIPTION
**Pull request by davidbenjamin**
@LeeTL1220 The first commit makes all concordance tools and MC3/M2 vcf merge much faster in Firecloud -- tasks that have been taking an hour will take a few minutes.  The second commit makes `GetPileupSummaries`, hence the contamination task much faster.  Previously that tool has cached all reads for the 100,000 bases around each site, so basically it had to do a whole bam's worth of I/O just to get ~60,000 pileups.

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/engine/AbstractConcordanceWalker.java [hunk 1]
```diff
@@ -77,7 +77,7 @@ private void initializeTruthVariantsIfNecessary() {
         }
 
         if (truthVariants == null) {
-            truthVariants = new FeatureDataSource<>(new FeatureInput<>(truthVariantsFile, "truth"), CACHE_LOOKAHEAD, VariantContext.class);
+            truthVariants = new FeatureDataSource<>(new FeatureInput<>(truthVariantsFile, "truth"), CACHE_LOOKAHEAD, VariantContext.class, cloudPrefetchBuffer, cloudIndexPrefetchBuffer);
```
Comment (cmnbroad) [inline src/main/java/org/broadinstitute/hellbender/engine/AbstractConcordanceWalker.java]: @davidbenjamin You should make this same change to the `evalVariants` `FeatureDataSource`.

Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/engine/AbstractConcordanceWalker.java]: Thanks @cmnbroad, did it.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/contamination/GetPileupSummaries.java [hunk 2]
```diff
@@ -55,14 +53,15 @@
  * gatk GetPileupSummaries \
  *   -I tumor.bam \
  *   -V common_biallelic.vcf.gz \
+ *   -V common_biallelic.vcf.gz \
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/contamination/GetPileupSummaries.java]: Why -V twice?

Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/contamination/GetPileupSummaries.java]: Whoops -- one is `-V` and one should be `-L`.  I have to admit this is kind of clumsy but it's not utterly redundant, at least.  For example, one might want to use some custom interval list (`-L`) of common sites but grab the allele frequencies from gnomAD (`-V`).

Fixed it and expanded the javadoc a bit.

**Comment by davidbenjamin** [comment]
@LeeTL1220 back to you.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/contamination/GetPileupSummaries.java [hunk 3]
```diff
@@ -65,6 +65,17 @@
  *   -O pileups.table
  * </pre>
  *
+ * Although the sites (-L) and variants (-V) resources will often be identical, this need not be the case.  For example,
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/contamination/GetPileupSummaries.java]: Typically these files would only be different if you were trying to use a subset of the variants?  I.e. -L was a subset of -V?  If so, can you add that to the doc?

Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/contamination/GetPileupSummaries.java]: Done.

**Pull request merged**